assets/js/blocks/mini-cart/edit.tsx

```diff
@@ -56,29 +49,28 @@ interface Props {
 	setProductCountColor: ( colorValue: string | undefined ) => void;
 }
 
-const Edit = ( {
-	attributes,
-	setAttributes,
-	clientId,
-	setPriceColor,
-	setIconColor,
-	setProductCountColor,
-}: Props ): ReactElement => {
+const defaultColorItem = {
+	name: undefined,
+	color: undefined,
+	slug: undefined,
+};
+
+const Edit = ( { attributes, setAttributes }: Props ): ReactElement => {
 	const {
 		cartAndCheckoutRenderStyle,
 		addToCartBehaviour,
 		hasHiddenPrice,
-		priceColorValue,
-		iconColorValue,
-		productCountColorValue,
+		priceColor = defaultColorItem,
+		iconColor = defaultColorItem,
+		productCountColor = defaultColorItem,
 		miniCartIcon,
 	} = attributes;
 
 	const className = classNames( {
 		'wc-block-mini-cart': true,
-		'has-price-color': priceColorValue,
-		'has-icon-color': iconColorValue,
-		'has-product-count-color': productCountColorValue,
+		'has-price-color': priceColor,
+		'has-icon-color': iconColor,
+		'has-product-count-color': productCountColor,
```

Comment:
This is an example of where a  `useColorPanelStyles` hook would be nice. However definitely not a must have at this point. Also, notice that the classnames here do not include the slug of the color (if it exists), which is what would be available with the generated classname in something like `priceColor.class` (if the slug exists - which it might not if the color picked by the user is a custom color).

Reply:
These class names will always be active since the `defaultColorItem` object will resolve as a truthy value. Is that expected behaviour even if no color has been selected?

cc @Aljullu since @nerrad is AFK.

Reply:
[I've fixed this](https://github.com/woocommerce/woocommerce-blocks/pull/10062/commits/8812b275310de3aae69c3b5cf9285f5b8f21e296), but I've also created the `useColorPanelStyles` hook at the same time which has reduced boilerplate code like @nerrad has pointed out. 

Currently the hook lives inside the same file as the ColorPanel component. I feel since they're tightly coupled this made sense but open to moving this.

Also the API of the hook returns an object structured like so `{ className: [ 'class-one', 'class-two' ] }` as a property on it. I've done this incase we decide to expand on it. It will reduce the need to refactor and retest other components that use this if we add anymore properties to this return value.